### PR TITLE
skip disks under scanning when healing disks

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
skip disks under scanning when healing disks

## Motivation and Context
Bonus: avoid calling DiskInfo() calls when missing blocks 
instead heal the object using MRF operation.

## How to test this PR?
The main motivation here is to save on IOPs

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by jiuker — APPROVED** [review]
code LGTM

File: cmd/global-heal.go [hunk 1]
```diff
@@ -342,11 +346,7 @@ func (er *erasureObjects) healErasureSet(ctx context.Context, buckets []string,
 				}
 				bgSeq.logHeal(madmin.HealItemObject)
 
-				select {
-				case <-ctx.Done():
-					return
-				case results <- result:
-				}
+				send(result)
```
Comment (klauspost) [inline cmd/global-heal.go]: This no longer returns when ctx.Done. Is that caught elsewhere?

Maybe return a bool from `send`.

**Review by klauspost — COMMENTED** [review]
Overall no objections.

File: cmd/global-heal.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/global-heal.go]: oh yes :-)

File: cmd/global-heal.go [hunk 2]
```diff
@@ -192,7 +192,7 @@ func (er *erasureObjects) healErasureSet(ctx context.Context, buckets []string,
 		tracker.setObject("")
 		tracker.setBucket(bucket)
 		// Heal current bucket again in case if it is failed
-		// in the  being of erasure set healing
+		// in the being of erasure set healing
```
Comment (krisis) [inline cmd/global-heal.go]:
```suggestion
		// in the beginning of erasure set healing
```

File: cmd/erasure.go [hunk 3]
```diff
@@ -317,9 +318,17 @@ func (er erasureObjects) getOnlineDisksWithHealing() (newDisks []StorageAPI, hea
...
 		}
-		newDisks = append(newDisks, disks[i])
+		if !info.Scanning {
+			newDisks = append(newDisks, disks[i])
+		} else {
+			scanningDisks = append(scanningDisks, disks[i])
+		}
 	}
 
+	// Add the disks being currently actively scanned to the
```
Comment (krisis) [inline cmd/erasure.go]:
```suggestion
	// Prefer new disks over disks which are currently being scanned. 
```

**Review by krisis — COMMENTED** [review]
LGTM. Minor nits and typo corrections suggested.

File: cmd/erasure.go [hunk 4]
```diff
@@ -317,9 +318,17 @@ func (er erasureObjects) getOnlineDisksWithHealing() (newDisks []StorageAPI, hea
...
-		newDisks = append(newDisks, disks[i])
+		if !info.Scanning {
+			newDisks = append(newDisks, disks[i])
+		} else {
+			scanningDisks = append(scanningDisks, disks[i])
+		}
 	}
 
+	// Prefer new disks over disks which are currently being scanned. 
+	// end to avoid them to be consider for listing first.
```
Comment (harshavardhana) [inline cmd/erasure.go]:
```suggestion
```

**Review by krisis — APPROVED** [review]

**Pull request merged**